hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/ArtifactManagementTest.java

```diff
@@ -173,14 +173,70 @@ public void createArtifactsUntilQuotaIsExceeded() throws NoSuchAlgorithmExceptio
                 .isEqualTo(maxArtifacts - 1);
 
         // now we should be able to create an artifact again
-        final byte random[] = RandomStringUtils.random(artifactSize).getBytes();
-        try (final InputStream inputStream = new ByteArrayInputStream(random)) {
+        final byte[] randomBytes = randomBytes(artifactSize);
+        try (final InputStream inputStream = new ByteArrayInputStream(randomBytes)) {
             artifactManagement.create(inputStream, sm1.getId(), "fileXYZ", false, artifactSize);
             assertThat(artifactRepository.findBySoftwareModuleId(PAGE, sm1.getId()).getTotalElements())
                     .isEqualTo(maxArtifacts);
         }
     }
 
+    @Test
+    @Description("Verifies that the quota specifying the maximum artifact storage is enforced (across software modules).")
+    public void createArtifactsUntilStorageQuotaIsExceeded() throws NoSuchAlgorithmException, IOException {
+
+        // create as many artifacts as possible w/o violating the storage quota
+        final long maxBytes = quotaManagement.getMaxArtifactStorage();
+        final List<Long> artifactIds = Lists.newArrayList();
+        final int artifactSize = 320 * 1024;
+        final int numArtifacts = Math.toIntExact(maxBytes / artifactSize);
+        for (int i = 0; i < numArtifacts; ++i) {
+            final JpaSoftwareModule sm = softwareModuleRepository
```

Comment:
I would extract the software module and artifact creation in its own private method, e.g. createSoftwareModuleWithArtifact(smId, artifactSize) that returns the artifact Id. Then you could reuse it later

Reply:
done